salt/minion.py

```diff
@@ -1253,7 +1253,8 @@ def __init__(
         # before we can get the grains.  We do this for proxies in the
         # post_master_init
         if not salt.utils.platform.is_proxy():
-            self.opts["grains"] = salt.loader.grains(opts)
```

Comment:
How does this fix the problem for "just" Windows?

Reply:
We only re-create the minion object for new jobs on windows. Yes, the change is taking place for all platforms and that is intentional.

https://github.com/saltstack/salt/pull/58903/files#diff-f694e1ed4605e0254070da941b4b86329c8d3f3462c7f117cccda51839e6d96dR1767

Reply:
The PR needs to reflect the change being made.   Pls remove "Windows" from the title.

Reply:
I was not clear before. This is only effecting windows. We don't need a windows specific gate for it.